sycl/include/sycl/ext/oneapi/weak_object.hpp

```diff
@@ -0,0 +1,128 @@
+//==-------------- weak_object.hpp --- SYCL weak objects -------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/buffer.hpp>
+#include <sycl/detail/defines_elementary.hpp>
+#include <sycl/ext/oneapi/weak_object_base.hpp>
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext {
+namespace oneapi {
+namespace detail {
+// Helper for creating ranges for empty weak_objects.
+template <int Dims> static range<Dims> createDummyRange() {
+  static_assert(Dims > 0 && Dims < 4, "Invalid dimensionality in range.");
```

Comment:
Why do we exclude 0? Isn't it used for scalars?

Reply:
My bad. I forget it is a thing. It has been added.